addons/transifex/models/transifex_code_translation.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import psycopg2
+
+from odoo import models, fields, _
+from odoo.tools.translate import CodeTranslations
+
+
+class TransifexCodeTranslation(models.Model):
+    _name = "transifex.code.translation"
+    _description = "Code Translation"
+    _log_access = False
+
+    source = fields.Text(string='Code')
+    value = fields.Text(string='Translation Value')
+    module = fields.Char(help="Module this term belongs to")
+    lang = fields.Selection(selection='_get_languages', string='Language', validate=False)
+    transifex_url = fields.Char("Transifex URL", compute='_compute_transifex_url',
+                                help="Propose a modification in the official version of Odoo")
+
+    def _auto_init(self):
+        super()._auto_init()
+        self.env.cr.execute(f'DELETE FROM {self._table}')
```

Comment:
Can you explain why the table must be emptied on module upgrade?

Reply:
Oh, I want to all data in the table to be cleared when a worker restarts.
Since the lifespan of the code translation should be the same as the worker. Every time, the worker restarts, the code translation in the po file might have been updated
But it seems `_auto_init` is a wrong hack point, since it requires the module is declared in `-i` or `-u` for `odoo-bin`
Do you have any suggestion to re-implement the logic?

Reply:
That looks like a very bad idea, because a server is not just _one_ worker.
If a worker cleans up the table when it loads the registry of the database, then it implies that the table will seem to be emptied _randomly_.  Think about a large database with 30+ workers!
I still don't understand why this should be synchronized with a worker's lifespan.

Reply:
Here is the flow we should support
1. a translator searches an untranslated code translation in the list view
2. the translator is redirected to the Transifex and translates the term
3. we update the code translations and restart the Odoo SaaS
4. the translator should see his translation is updated in the list view

Code translations
1. are static and loaded on demand and won't be cleaned within the lifespan of a process/worker
2. might be different after we restart a process/worker since the po file may have be updated

Here I assume all workers will be started/stopped at the same time
I hope every time a worker is restarted, the data for code translation will be cleaned
And I thought `Delete` everytime for an empty table is very fast

Reply:
> Here I assume all workers will be started/stopped at the same time

You may not make that assumption: workers are stopped and respawned when they reach a certain limit (typically a number of requests).